src/commands/run.rs

```diff
@@ -526,11 +526,15 @@ impl RunCommand {
                 .to_str()
                 .ok_or_else(|| anyhow!("argument is not valid utf-8: {val:?}"))?;
             values.push(match ty {
-                // TODO: integer parsing here should handle hexadecimal notation
-                // like `0x0...`, but the Rust standard library currently only
-                // parses base-10 representations.
-                ValType::I32 => Val::I32(val.parse()?),
-                ValType::I64 => Val::I64(val.parse()?),
+                // Supports both decimal and hexadecimal notation (with 0x prefix)
+                ValType::I32 => Val::I32(
+                    val.parse::<i32>()
+                        .or_else(|_| i32::from_str_radix(&val.trim_start_matches("0x"), 16))?,
```

Comment:
This unfortunately has the behavior that `ff` is interpreted as 255 since the `trim_start_matches` is an optional trim rather than a required trim. Could this be refactored to require the `0x` prefix for hexadecimal arguments?